papers/tests/test_pages.py

```diff
@@ -64,6 +66,24 @@ def checkHtml(self, resp):
             print(resp.content)
             print("HTML validation error: "+str(e))
```

Comment:
How does your new validation code below compare with the one we had above? If it is stricter, then can we remove the previous checks above, and drop the dependency to html5lib?

Reply:
I think this is a superset of the checks made by `html5lib` indeed. `html5lib` is trying to parse the HTML and will fail on invalid (in the sense of not parsable) HTML.

My test will consider additional checks such as invalid attributes or this kind of thing.

However, I'm wondering whether we should not in the end stick with `html5lib` and use the W3C Validator API to check for full HTML validity, I'm not 100% confident `html-tidy` is doing everything right.

Reply:
Actually, the W3C validator can run locally (https://github.com/validator/validator)! I'm using it (actually the Python `html5validator` wrapper).

Bonus points, I found a couple of small display issues like this (such as the labels not working as expected in the upload form) :)